arangod/GeneralServer/AsyncJobManager.cpp

```diff
@@ -277,7 +278,7 @@ std::vector<AsyncJobResult::IdType> AsyncJobManager::byStatus(
 /// @brief initializes an async job
 ////////////////////////////////////////////////////////////////////////////////
 
-void AsyncJobManager::initAsyncJob(RestHandler* handler) {
+void AsyncJobManager::initAsyncJob(std::shared_ptr<RestHandler> handler) {
```

Comment:
A few lines down - adapt to allow moving `handler` in the call of the `AsyncJobResult` constructor:
```
   std::string user = handler->request()->user();
   AsyncJobResult ajr(jobId, AsyncJobResult::JOB_PENDING, std::move(handler));
```